From a0270dbb9becc900e2fee978eeeb9e24ee37fdf0 Mon Sep 17 00:00:00 2001
From: Sven Roederer <freifunk@it-solutions.geroedel.de>
Date: Tue, 26 Jun 2018 00:33:22 +0200
Subject: [PATCH] olsrd-defaults: dont use cron to reload dnsmasq

* defaults-olsrd: use sighup-pid-file of nameservice-plugin
  * this seems to be a more elegant way to have dnsmasq reread the hosts-file
* dhcp-defaults: remove obsolete dnsmasq restart by cron
* olsrd-defaults: use nameservice/name-change-script to reload dnsmasq
  * This avoids the use of the dynamically created pid-filename
---
 defaults/freifunk-berlin-dhcp-defaults/Makefile    |  1 -
 .../files/etc/init.d/ff_dnsmasq_reload             | 14 --------------
 defaults/freifunk-berlin-olsrd-defaults/Makefile   |  4 ++--
 .../uci-defaults/freifunk-berlin-olsrd-defaults    |  4 ++++
 4 files changed, 6 insertions(+), 17 deletions(-)
 delete mode 100755 defaults/freifunk-berlin-dhcp-defaults/files/etc/init.d/ff_dnsmasq_reload

diff --git a/defaults/freifunk-berlin-dhcp-defaults/Makefile b/defaults/freifunk-berlin-dhcp-defaults/Makefile
index 6e5daf788..5f9e67ab8 100644
--- a/defaults/freifunk-berlin-dhcp-defaults/Makefile
+++ b/defaults/freifunk-berlin-dhcp-defaults/Makefile
@@ -34,7 +34,6 @@ endef
 define Package/freifunk-berlin-dhcp-defaults/install
 	$(INSTALL_DIR) $(1)/etc/uci-defaults
 	$(CP) ./uci-defaults/* $(1)/etc/uci-defaults
-	$(CP) ./files/* $(1)/
 endef
 
 $(eval $(call BuildPackage,freifunk-berlin-dhcp-defaults))
diff --git a/defaults/freifunk-berlin-dhcp-defaults/files/etc/init.d/ff_dnsmasq_reload b/defaults/freifunk-berlin-dhcp-defaults/files/etc/init.d/ff_dnsmasq_reload
deleted file mode 100755
index 95a312153..000000000
--- a/defaults/freifunk-berlin-dhcp-defaults/files/etc/init.d/ff_dnsmasq_reload
+++ /dev/null
@@ -1,14 +0,0 @@
-#!/bin/sh /etc/rc.common
-
-START=88
-
-boot() {
-	test -f /etc/crontabs/root || touch /etc/crontabs/root
-
-	grep -q 'killall -HUP dnsmasq' /etc/crontabs/root || {
-		echo "*/5 * * * *	killall -HUP dnsmasq" >> /etc/crontabs/root
-	}
-
-	/etc/init.d/cron restart &
-}
-
diff --git a/defaults/freifunk-berlin-olsrd-defaults/uci-defaults/freifunk-berlin-olsrd-defaults b/defaults/freifunk-berlin-olsrd-defaults/uci-defaults/freifunk-berlin-olsrd-defaults
index 8e1ff4540..02ffd5bd1 100644
--- a/defaults/freifunk-berlin-olsrd-defaults/uci-defaults/freifunk-berlin-olsrd-defaults
+++ b/defaults/freifunk-berlin-olsrd-defaults/uci-defaults/freifunk-berlin-olsrd-defaults
@@ -24,6 +24,8 @@ uci set olsrd.$PLUGIN.suffix=.olsr
 uci set olsrd.$PLUGIN.hosts_file=/tmp/hosts/olsr
 uci set olsrd.$PLUGIN.latlon_file=/var/run/latlon.js
 uci set olsrd.$PLUGIN.services_file=/var/etc/services.olsr
+uci set olsrd.$PLUGIN.name_change_script="/etc/init.d/dnsmasq reload"
+uci set olsrd.$PLUGIN.filewrite_interval=60
 
 # add jsoninfo plugin
 PLUGIN="$(uci add olsrd LoadPlugin)"
@@ -99,6 +101,8 @@ uci set olsrd6.$PLUGIN.suffix=.olsr
 uci set olsrd6.$PLUGIN.hosts_file=/tmp/hosts/olsr.ipv6
 uci set olsrd6.$PLUGIN.latlon_file=/var/run/latlon.js.ipv6
 uci set olsrd6.$PLUGIN.services_file=/var/etc/services.olsr.ipv6
+uci set olsrd6.$PLUGIN.name_change_script="/etc/init.d/dnsmasq reload"
+uci set olsrd6.$PLUGIN.filewrite_interval=60
 
 # add jsoninfo plugin
 PLUGIN="$(uci add olsrd6 LoadPlugin)"
-- 
2.20.1

